packages/framework/src/Component/Maintenance/MaintenanceModeFacade.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrameworkBundle\Component\Maintenance;
+
+use Shopsys\FrameworkBundle\Component\Redis\RedisClientFacade;
+
+class MaintenanceModeFacade
+{
+    protected ?bool $isMaintenanceModeEnabled = null;
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\Redis\RedisClientFacade $redisClientFacade
+     */
+    public function __construct(
+        protected readonly RedisClientFacade $redisClientFacade,
+    ) {
+    }
+
+    public function disable(): void
+    {
+        $this->redisClientFacade->save(MaintenanceModeSubscriber::MAINTENANCE_KEY, true);
+        $this->isMaintenanceModeEnabled = true;
+    }
+
+    public function enable(): void
+    {
+        $this->redisClientFacade->delete(MaintenanceModeSubscriber::MAINTENANCE_KEY);
+        $this->isMaintenanceModeEnabled = false;
```

Comment:
function `enable` sets `isEnabled` to false - it seems counterintuitive

Reply:
🤦